### DISCUSSION TITLE
ClinVar Annotation Updates for seqr local installations

### DISCUSSION THREAD

**Discussion by anjalijain22** [broadinstitute/seqr#3370] [Q&A] [answered]
Hi,

Recently, our team noticed some discrepancies between the ClinVar annotations shown on seqr versus annotations on ClinVar. For example: This variant is annotated as "likely pathogenic" on seqr:
<img width="1268" alt="Screen Shot 2023-05-15 at 9 48 39 AM" src="https://github.com/broadinstitute/seqr/assets/60684431/69b70f2d-efde-4520-a874-0d9ef01a6999">

But on ClinVar, it is annotated as likely Benign: https://www.ncbi.nlm.nih.gov/clinvar/variation/986331/?oq=974463[alleleid]&m=NM_014585.6(SLC40A1):c.689C%3EA%20(p.Thr230Asn) 
<img width="1279" alt="Screen Shot 2023-05-15 at 9 51 24 AM" src="https://github.com/broadinstitute/seqr/assets/60684431/697942b3-d069-44b5-b87d-be43029d6a10">

I suspect this is because the "likely benign" submission to ClinVar was done relatively recently (Oct 19, 2022) and we are using an older version of the ClinVar Hail matrix table from where seqr fetches its ClinVar annotations. So I am wondering how can we update the ClinVar matrix table for local installations? 

Many thanks for your help, appreciate it!

Anjali

**Comment by hanars** [comment, marked as answer]
You are correct that issue arises from using an older clinvar matrix table. To fix it, you first need to download the updated table- this will depend on whether or not you are using google cloud or not:

To local files:
```
docker-compose exec pipeline-runner rm -rf "/seqr-reference-data/GRCh${BUILD_VERSION}/clinvar.GRCh${BUILD_VERSION}.ht"
docker-compose exec pipeline-runner gsutil -m rsync -r "gs://seqr-reference-data/GRCh${BUILD_VERSION}/clinvar/clinvar.GRCh${BUILD_VERSION}.ht" "/seqr-reference-data/GRCh${BUILD_VERSION}/clinvar.GRCh${BUILD_VERSION}.ht"
```

For GCP/dataproc:
```
gsutil rm -r "${GS_BUCKET}/reference_data/GRCh${BUILD_VERSION}/clinvar.GRCh${BUILD_VERSION}.ht"
gsutil rsync -r "gs://seqr-reference-data/GRCh${BUILD_VERSION}/clinvar/clinvar.GRCh${BUILD_VERSION}.ht" "${GS_BUCKET}/reference_data/GRCh${BUILD_VERSION}/clinvar.GRCh${BUILD_VERSION}.ht"
```

You will then need to re-load your projects and add the new indices to seqr in order to get the updated annotations in your search results

**Reply by anjalijain22** [reply]
Thank you for your quick response, Hana! We will give this a try and let you know if it worked.

I'm curious to know how often are the ClinVar annotations as well as other databases (omim, clingen, refseq etc) updated and how often do you recommend we should be running these commands to update our local installations?

**Reply by hanars** [reply]
So there are 2 types of reference data in seqr - those that are annotated during data loading and those that are stored directly in seqr. Data like clinvar that is added during data loading are harder to update, as in addition to updating the data source you also have to re-load all your data using the full loading pipeline. For that reference data, we recommend timing your updates around planned data loading and re-anlaysis. If you have a large amount of data thats going to be added to seqr, you should update the reference data before you do that loading. If you have projects that never get new samples and thus are never reloaded, its probably worth updating the reference data and reloading those when you have the bandwidth to go back and re-analyse those cases, maybe after a year. 

For omim, clingen etc, that data is stored directly in seqr and is pretty straightforward to update and does not require any reloading. We update that weekly using an automated cron job,  but you could also run it weekly or monthly or whatever is convenient. You can do that using the following command:
```
docker-compose exec seqr manage.py update_all_reference_data --use-cached-omim --skip-gencode
```

**Reply by anjalijain22** [reply]
I see, that makes sense! Thanks for explaining!

From the[ update_all_reference_data.py](https://github.com/broadinstitute/seqr/blob/master/reference_data/management/commands/update_all_reference_data.py) script, I understand that OMIM, ClinGen, MGI, dbnsfp, primate_ai, gencode, refseq and HPO are the datasets that are stored directly in seqr. In terms of data that is annotated during data loading, what other data apart from ClinVar falls in that category? We're doing a big data intake over the next few weeks so I want to ensure that we're using the most updated set of annotations.

**Reply by hanars** [reply]
So other than clinvar, everything else annotated during data loading is contained in a single combined_reference_data table. You can update it using the same instructions as above for updating clinvar, except instead of `clinvar.GRCh${BUILD_VERSION}.ht` as the file path use `combined_reference_data_grch${BUILD_VERSION}.ht`

**Reply by anjalijain22** [reply]
That makes sense! We'll update that on our end as well. How often are these hail tables updated on your end (monthly, once every 6 months, yearly)?

Also going back to the original issue, we updated the ClinVar Hail table and now the variant is correctly annotated as likely benign :) Is it possible to add the update commands for the reference databases in the README for local installs? I believe this will be very helpful.

**Reply by anjalijain22** [reply]
Another thing that I just thought of is - in the event that we do reload data for certain projects for reanalysis, we would lose all the previous variant tags/notes etc done on the previous data loaded for that family, correct?

**Reply by hanars** [reply]
We update the annotation tables roughly every 6 months, or whenever we have a new annotation we want to add (maybe once or twice a year). 

> in the event that we do reload data for certain projects for reanalysis, we would lose all the previous variant tags/notes etc done on the previous data loaded for that family, correct?

That is incorrect. Tags and notes will remain! seqr will attempt to update the variant to use the latest annotations, so for example if you had previously tagged that "pathogenic" variant, after reloading the project when you go to the saved variant page the variant will be there with all the tags and notes but should be showing as likely benign now

**Reply by anjalijain22** [reply]
Oh I see, that's great to know!!

Thank you so much Hana for all your help with this, much appreciated!
